core/admin/mailu/models.py

```diff
@@ -261,14 +261,23 @@ def resolve_domain(cls, email):
     @classmethod
     def resolve_destination(cls, localpart, domain_name, ignore_forward_keep=False):
         localpart_stripped = None
+        stripped_alias = None
+
         if os.environ.get('RECIPIENT_DELIMITER') in localpart:
             localpart_stripped = localpart.rsplit(os.environ.get('RECIPIENT_DELIMITER'), 1)[0]
 
-        alias = Alias.resolve(localpart, domain_name)
-        if not alias and localpart_stripped:
-            alias = Alias.resolve(localpart_stripped, domain_name)
-        if alias:
-            return alias.destination
+        pure_alias = Alias.resolve(localpart, domain_name)
+        has_wildcard = pure_alias and '%' in pure_alias.email and pure_alias.wildcard
+        stripped_alias = Alias.resolve(localpart_stripped, domain_name)
+
+        if pure_alias and not has_wildcard:
+            return pure_alias.destination
+        elif not pure_alias and stripped_alias:
+            return stripped_alias.destination
+        elif has_wildcard and localpart_stripped and stripped_alias:
+            return stripped_alias.destination
+        elif pure_alias:
+            return pure_alias.destination
```

Comment:
This seems way to complicated to me. Why we don't just always split the localpart at the recipient delimiter and make one alias lookup? 

With your example from #813: 
- `foo@xxx.tld` ⇒ User A
- `%@xxx.tld` ⇒ User B

If we now send an email to ` foo+spam@xxx.tld` after the split we look for the alias `foo@xxx.tld`. And if someone sends directly to `foo@xxx.tld` the split doesn't change anything and we resolve `foo@xxx.tld`. 
So we could reduce this to a few lines of code and do just one database lookup.

Or am I missing here something? :man_shrugging:

Reply:
My thought was that maybe a user could want to re-route `foo+spam@xxx.tld` a a later time due to his/her changed personal preferences … then an alias like that should take precedence over `alias foo with delimiter stripped`. If my thought on this is a bit far-fetched, I’ll gladly build it to look easier again.

Reply:
I think we shouldn't allow this kind of alias. Postfix doesn't allow the delimiter in aliases too. So imo we should do it the "postfix-way".